src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -80,30 +83,35 @@ public async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdentit
             bool includePrerelease, 
             CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             // get all package references for all the projects and cache locally
             var packageReferences = await project.GetInstalledPackagesAsync(cancellationToken);
 
             // filter package references for current package identity
             var matchedPackageReferences = packageReferences
                 .Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.PackageIdentity.Id, identity.Id));
 
+            cancellationToken.ThrowIfCancellationRequested();
             // Allowed version range for current package across all selected projects
             // Picks the first non-default range
             var allowedVersions = matchedPackageReferences
                 .Select(r => r.AllowedVersions)
                 .FirstOrDefault(v => v != null) ?? VersionRange.All;
 
+            cancellationToken.ThrowIfCancellationRequested();
             var tasks = _sourceRepositories
                 .Select(r => GetMetadataTaskSafeAsync(() => r.GetLatestPackageMetadataAsync(identity.Id, includePrerelease, cancellationToken, allowedVersions)))
                 .ToArray();
 
             var completed = (await Task.WhenAll(tasks))
                 .Where(m => m != null);
 
+            cancellationToken.ThrowIfCancellationRequested();
             var highest = completed
                 .OrderByDescending(e => e.Identity.Version, VersionComparer.VersionRelease)
                 .FirstOrDefault();
 
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Similar to my previous comment, calling `ThrowIfCancellationRequested` 5 times in this method feels a bit much. Perhaps my expectations are unrealistic, but even if there are a large number of package sources, I would expect the CPU bound portions of this method to finish very quickly, especially since we fixed [the issue of a package source returning more results than requested](https://github.com/NuGet/NuGet.Client/pull/3406).

Reply:
Explained https://github.com/NuGet/NuGet.Client/pull/3436#discussion_r437482637
This case is just LINQ, so I'll scale these calls back at least by 1.